table/column.go

```diff
@@ -278,6 +278,12 @@ func handleZeroDatetime(ctx sessionctx.Context, col *model.ColumnInfo, casted ty
 // TODO: change the third arg to TypeField. Not pass ColumnInfo.
 func CastValue(ctx sessionctx.Context, val types.Datum, col *model.ColumnInfo, returnErr, forceIgnoreTruncate bool) (casted types.Datum, err error) {
 	sc := ctx.GetSessionVars().StmtCtx
+	if val.Kind() == types.KindNull && col.FieldType.Tp == mysql.TypeTimestamp && ctx.GetSessionVars().StmtCtx.IsDDLJobInQueue {
```

Comment:
Could we put this logic out of `CastValue` and before it in `getRowRecord`? Then we needn't affect other function who calls `CastValue`.